frontend/src/components/game/Editor.tsx

```diff
@@ -70,34 +30,41 @@ function ResizableMonacoEditor(props: EditorProps) {
     });
   };
 
-  const handleLanguageChange = (language: string) => {
+  const handleLanguageChange = (language: Language) => {
     // Save the code for this language
-    languages[currentLanguage].defaultCode = codeEditor!.getValue();
+    if (codeMap != null && codeEditor != null) {
+      codeMap[currentLanguage] = codeEditor.getValue();
+      codeEditor.setValue(codeMap[language]);
```

Comment:
Note: if language is not in the map, I get this error in the console: 

<img width="666" alt="Screenshot" src="https://user-images.githubusercontent.com/24768574/106425988-a4fdf980-6419-11eb-986c-53699bd26f29.png">

However, the default behavior is that the code in the editor simply stays as is, so think it's fine for now, just something to note

Reply:
Yes, this error will go away once we implement all of the languages 👍